File: chaerim/Programmers/Level_2/Lv2_할인_행사.py
def solution(want, number, discount):
    answer = 0
    
    want_dict = {item: cnt for item, cnt in zip(want, number)}
    
    for day in range(len(discount) - 9):
        discounted_items = {}
        for i in range(day, day + 10):
            if discount[i] in discounted_items:
                discounted_items[discount[i]] += 1
            else:
                discounted_items[discount[i]] = 1
        
        if want_dict == discounted_items:
            answer += 1

    return answer

File: seongwon/Programmers/Level_2/Lv2_할인_행사.cpp
#include <string>
#include <vector>

using namespace std;

int solution(vector<string> want, vector<int> number, vector<string> discount) {
    int answer = 0;
    for(int i = 0; i < discount.size()-9; i++){
        vector<int>cnt = number;
        for(int j = i; j < i+10; j++){
            for(int k = 0; k < want.size(); k++){
                if(want[k] == discount[j]){
                    cnt[k]--;
                }
            }
        }
        vector<int> t(number.size(), 0);
        if(t == cnt)
            answer++;
    }
    return answer;
}
